sentinel-extension/sentinel-annotation-aspectj/src/main/java/com/alibaba/csp/sentinel/annotation/aspectj/AbstractSentinelAspectSupport.java

```diff
@@ -149,21 +133,48 @@ protected Object handleBlockException(ProceedingJoinPoint pjp, SentinelResource
             // Construct args.
             Object[] args = Arrays.copyOf(originArgs, originArgs.length + 1);
             args[args.length - 1] = ex;
-            try {
-                if (isStatic(blockHandlerMethod)) {
-                    return blockHandlerMethod.invoke(null, args);
-                }
-                return blockHandlerMethod.invoke(pjp.getTarget(), args);
-            } catch (InvocationTargetException e) {
-                // throw the actual exception
-                throw e.getTargetException();
-            }
+            return invoke(pjp, blockHandlerMethod, args);
         }
 
         // If no block handler is present, then go to fallback.
         return handleFallback(pjp, annotation, ex);
     }
 
+    private Object invoke(ProceedingJoinPoint pjp, Method method, Object[] args) throws Throwable {
+        boolean accessible = method.isAccessible();
+        try {
+            if (!accessible) {
+                makeAccessible(method);
+            }
+            if (isStatic(method)) {
+                return method.invoke(null, args);
+            }
+            return method.invoke(pjp.getTarget(), args);
+        } catch (InvocationTargetException e) {
+            // throw the actual exception
+            throw e.getTargetException();
+        } finally {
+            if (!accessible) {
+                method.setAccessible(false);
+            }
+        }
+    }
+
+    /**
+     * Make the given method accessible, explicitly setting it accessible if
+     * necessary. The {@code setAccessible(true)} method is only called
+     * when actually necessary, to avoid unnecessary conflicts with a JVM
+     * SecurityManager (if active).
+     * @param method the method to make accessible
+     * @see java.lang.reflect.Method#setAccessible
+     */
+    private static void makeAccessible(Method method) {
+        if ((!Modifier.isPublic(method.getModifiers()) ||
```

Comment:
@wuwen5 can we make the expression simple? like assign the first judge to a variable, and add the explain to it.

Reply:
OK, now it has been modified. the method is copy from spring-core's ```ReflectionUtils```

Reply:
Nice work!